### ISSUE TITLE
Db2Catalog can skip packages with identical LastEdited timestamps

### ISSUE THREAD

**Issue by joelverhagen** [NuGet/NuGetGallery#2841]
I found this in the DEV environment.

Suppose two packages are published at the same time, down to the millisecond. The feed2catalog job uses the package Created date as an exclusive cursor across the feed (i.e. `where Created > {last Created seen}`, not `where Created >= {last Created seen}`). The feed2catalog job processes in batches of 20 packages.

This is a trimmed down version of the feed query made by the feed2catalog job:
`/api/v2/Packages()?$filter=Created gt datetime'2015-12-28T23:32:36.627'&$orderby=Created&$select=Created`

Make note of the `2015-12-28T23:32:36.627` parameter in the URL. This query returns three packages with the following `Created` dates:
1. 2015-12-28T23:32:36.64
2. 2015-12-28T23:32:36.64
3. 2015-12-28T23:32:36.65

In this particular instance, a query executed by the job returned items up to and including item 1. This means item 1 was processed and entered into the catalog. When the job went for the next batch, it did a query `where Created > '2015-12-28T23:32:36.64'`, which of course excludes item 2 and includes item 3, leading to a gap in the catalog with respect to the feed.

One possible fix is to switch to `>=` (`ge` in OData) and ensure that duplicate items are managed. Another possibility is to investigate using [`TOP WITH TIES`](https://msdn.microsoft.com/en-us/library/ms189463.aspx) at the SQL layer (not sure how this jives with OData).

@johnataylor, @RanjiniM any thoughts on this? I have also heard that clocks can be adjusted backwards by the OS sometimes so assuming absolute order of `Created` may not be the best approach.

**Comment by maartenba** [comment]
My vote is on  `>=`, just need to ensure no packages are handled multiple times in that case.

**Comment by joelverhagen** [comment]
The (perhaps theoretical) caveat with that solution is that if page size is N and N items have the exact same created date, we would get stuck. If we did `>=` in conjunction with `skip`, I think we'd be fine.

**Comment by maartenba** [comment]
Yeah perhaps use the same base date and run a few skips until no results left or the date actually increases?

**Comment by maartenba** [comment]
Seems for this to work properly, we need to store the $skip value in the cursor as well.

**Comment by agr** [comment]
I'm not sure DB guarantees that record `2` cannot appear in DB after record `3` (unless we run read transaction with `SERIALIZABLE` [isolation level](https://technet.microsoft.com/en-us/library/ms173763(v=sql.105).aspx), doing this though would likely incur heavy perf penalties).
And I don't have any good ideas about how to work around that...

**Comment by scottbommarito** [comment]
This bug seems to be a big blocker on https://github.com/nuget/engineering/issues/2082
